modules/opengl/src/canvasprocessorgl.cpp

```diff
@@ -44,4 +47,17 @@ const ProcessorInfo CanvasProcessorGL::getProcessorInfo() const { return process
 
 CanvasProcessorGL::CanvasProcessorGL(InviwoApplication* app) : CanvasProcessor(app) {}
 
+void CanvasProcessorGL::process() {
+    // ensure that the image inport has a GL representation
+    // otherwise the canvas widget will request a new one which in turn creates an fbo within the
+    // context of the canvas, i.e. not the default canvas.
+    //
+    // FIXME: this is not ideal
+    if (!inport_.getData()->hasRepresentation<ImageGL>()) {
+        RenderContext::getPtr()->activateDefaultRenderContext();
```

Comment:
No need to activate default RenderContext? Should already be active since we are in the process function?

Reply:
That is exactly the point. In same borderline cases, the gl widget is created just before this call. Hence activating its context.